astropy/modeling/core.py

```diff
@@ -1626,7 +1649,7 @@ def rename(self, name):
         new_model._name = name
         return new_model
 
-    @sharedmethod
+    @property
     def n_submodels(self):
```

Comment:
This change is pretty significant and should definitely be mentioned in the API changes docs?

Reply:
Agreed. A page with changes is coming after this PR.